pkg/netceptor/tlsconfig.go

```diff
@@ -65,7 +79,16 @@ func (cfg TLSServerCfg) Prepare() error {
 		tlscfg.ClientAuth = tls.NoClientCert
 	}
 
-	return MainInstance.SetServerTLSConfig(cfg.Name, tlscfg)
+	serverConf := tlscfg
+	if cfg.VerifyClientNodeID {
+		tlscfg.NextProtos = []string{"netceptor"}
+		serverConf = &tls.Config{}
```

Comment:
Maybe I'm misunderstanding this, but it seems to be throwing away the previous settings in serverConf?

Reply:
yea in the case of `VerifyClientNodeID`, we set serverConf to an empty tls.Config{} and only set the `GetConfigForClient` field, which ultimately returns the tlscfg we just defined.

if it is more clear I can do an if / else

```golang
if cfg.VerifyClientNodeID {
	tlscfg.NextProtos = []string{"netceptor"}
	serverConf = &tls.Config{}
	serverConf.GetConfigForClient = func(hi *tls.ClientHelloInfo) (*tls.Config, error) {
		tlscfg.VerifyPeerCertificate = getClientValidator(hi, tlscfg.ClientCAs)
		return tlscfg, nil
	}
} else {
	serverConf := tlscfg
}
return MainInstance.SetServerTLSConfig(cfg.Name, serverConf)
```

Reply:
Doesn't this drop the server's certificates when VerifyClientNodeID is true?

Reply:
the `GetConfigForClient` callback returns `tlscfg`, which will have `Certificates` as defined on line 62.
When `VerifyClientNodeID` is set, nothing is dropped from tlscfg, but `VerifyPeerCertificate` and `NextProtos` are added